src/prefect/tasks/cubejs/cubejs_tasks.py

```diff
@@ -0,0 +1,187 @@
+                Query format can be found at: https://cube.dev/docs/query-format
+            - security_context (str, dict, optional): The security context to use
+                during authentication.
+                If the security context does not contain an expiration period,
+                then a 7 days expiration period is added automatically.
+                More info at: https://cube.dev/docs/security/context
+
+        Raises:
+            - ValueError if both `subdomain` and `url` are missing.
+            - ValueError if `api_token` is missing and `api_token_env_var` cannot be found.
+            - ValueError if `query` is missing.
+            - `prefect.engine.signals.FAIL` if the CubeJS Query API fails.
+
+        Returns:
+            - The CubeJS JSON response
+
+        """
+
+        if not subdomain and not url:
+            raise ValueError("Missing both `subdomain` and `url`.")
+
+        if not api_secret and api_secret_env_var not in os.environ:
+            raise ValueError("Missing `api_secret` and `api_secret_env_var` not found.")
+
+        if not query:
+            raise ValueError("Missing `query`.")
+
+        cube_base_url = self.__CUBEJS_CLOUD_BASE_URL
+        if subdomain:
+            cube_base_url = f"{cube_base_url.format(subdomain=subdomain)}/cubejs-api"
+        else:
+            cube_base_url = url
+        query_api_url = f"{cube_base_url}/v1/load"
+
+        self.logger.debug(f"Query URL: {query_api_url}")
+
+        secret = api_secret if api_secret else os.environ[api_secret_env_var]
+
+        if security_context:
+
+            extended_context = security_context
+            if "exp" not in security_context and "expiresIn" not in security_context:
+                extended_context["epxiresIn"] = "7d"
+            api_token = jwt.encode(
+                payload=extended_context, key=secret, algorithm="HS256"
+            )
+
+            self.logger.debug("JWT token generated with security context.")
+
+        else:
+            api_token = jwt.encode(payload={}, key=secret)
+
+        session = Session()
+        session.headers = {
+            "Content-type": "application/json",
+            "Authorization": api_token,
+        }
+
+        params = {"query": json.dumps(query)}
+
+        secs = 2
+        while True:
+            with session.get(url=query_api_url, params=params) as response:
+
+                self.logger.debug(f"URL is: {response.url}")
+
+                if response.status_code == 200:
+                    data = response.json()
+
+                    if "error" in data.keys() and "Continue wait" in data["error"]:
+                        self.logger.warning(
+                            f"CubeJS Query still running. Waiting {secs} seconds before retrying..."
+                        )
+                        time.sleep(secs)
+                        secs *= 2 if secs < 60 else 2
+                        continue
```

Comment:
you could add a parameter to tune the number of subsequent retries and raise in case of counted failures reach the ceil

Reply:
I've added a `max_wait_time` parameter, similarly to what has been implemented in `DbtCloudRunJob` task.